app/models/piece.rb

```diff
@@ -0,0 +1,7 @@
+class Piece < ActiveRecord::Base
```

Comment:
I don't follow. My understanding is that Rails should take care of all the STI scoping automatically, as long as the models inherit properly from Piece and have class names that match up with the values in the `type` column. That's why they're all capitalized. 

I tested it in dev, and it seemed to work fine.